cmd/kubectl-gadget/snapshot/snapshot.go

```diff
@@ -15,16 +15,118 @@
 package snapshot
 
 import (
-	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
 
 	"github.com/spf13/cobra"
-)
 
-// All the gadgets within this package use this global variable, so let's
-// declare it here.
-var params utils.CommonFlags
+	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/apis/gadget/v1alpha1"
+	processcollectortypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/process-collector/types"
+	socketcollectortypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/socket-collector/types"
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+)
 
 var SnapshotCmd = &cobra.Command{
 	Use:   "snapshot",
 	Short: "Take a snapshot of a subsystem and print it",
 }
+
+type GadgetEvent interface {
+	socketcollectortypes.Event | processcollectortypes.Event
+
+	// The Go compiler does not support accessing a struct field x.f where x is
+	// of type parameter type even if all types in the type parameter's type set
+	// have a field f. We may remove this restriction in Go 1.19. See
+	// https://tip.golang.org/doc/go1.18#generics.
+	GetBaseEvent() eventtypes.Event
+}
+
+type GadgetFlags interface {
+	SocketFlags | ProcessFlags
+}
```

Comment:
Defining the [`processFlags`](https://github.com/kinvolk/inspektor-gadget/blob/e9cc720de5e12bf926bebc42a69417afe4959d31/cmd/kubectl-gadget/snapshot/process.go#L45) and [`socketFlags`](https://github.com/kinvolk/inspektor-gadget/blob/e9cc720de5e12bf926bebc42a69417afe4959d31/cmd/kubectl-gadget/snapshot/socket.go#L46) as global variables would remove the need of having this interface. On the one hand, I would prefer not to use global variables unless it is strictly needed. However, I am aware that this interface makes the code more complex. What do you think?

Reply:
Using an interface resolved this situation.